db_test.go

```diff
@@ -900,6 +902,96 @@ func TestWithContextDoesNotMutate(t *testing.T) {
 	}
 }
 
+func TestWithDeadline(t *testing.T) {
+	db := makeTestDB(t, usersDDL)
+	db.DeadlineQueryRewriter = func(db *DB, query interface{}, millis int64) (interface{}, error) {
+		serializer, err := db.GetSerializer(query)
```

Comment:
Would all implementations depend only on the serialized query? If so, could you change the rewriter interface to only pass in the query string? That would allow you to avoid making `getSerializer()` public.